libs/gltfio/src/AssetLoader.cpp

```diff
@@ -444,22 +444,20 @@ void FAssetLoader::createRenderable(const cgltf_data* srcAsset, const cgltf_node
 
     Aabb aabb;
 
-    // glTF spec says that all primitives MUST have the same number of morph targets in the same order.
-    const cgltf_size numMorphTargets = mesh->weights_count;
-    builder.morphing(numMorphTargets);
-
+    cgltf_size numMorphTargets = 0;
     // For each prim, create a Filament VertexBuffer, IndexBuffer, and MaterialInstance.
     for (cgltf_size index = 0; index < nprims; ++index, ++outputPrim, ++inputPrim) {
         RenderableManager::PrimitiveType primType;
         if (!getPrimitiveType(inputPrim->type, &primType)) {
             slog.e << "Unsupported primitive type in " << name << io::endl;
         }
 
-        if (numMorphTargets > 0 && inputPrim->targets_count != numMorphTargets) {
-            slog.e << "Sister primitives must all have the same number of morph targets."
-                   << io::endl;
-            mError = true;
-            continue;
+        if (inputPrim->targets_count > 0) {
```

Comment:
I don't think this will catch cases where some primitives have no morph targets, but others do. Is that allowed? For example:
```
Primitive 0: 0 morph targets
Primitive 1: 5 morph targets
Primitive 2: 5 morph targets
```

Reply:
This case is invalid glTF spec so it must be not allowed.

Reply:
My point is that the scenario where the first primitive has no morph targets (but other primitives do) will _not_ be reported as "Sister primitives must all have the same number of morph targets." It's an invalid glTF file, yes, but so are the ones where we show that message. We should be consistent.

```
            "primitives" : [
                {
                    "attributes" : {
                        "POSITION" : 0,
                        "NORMAL" : 1,
                        "TEXCOORD_0" : 2
                    },
                    "indices" : 3,
                    "material" : 0
                    "targets" : [
                    ]
                },
                {
                    "attributes" : {
                        "POSITION" : 8,
                        "NORMAL" : 9,
                        "TEXCOORD_0" : 10
                    },
                    "indices" : 3,
                    "material" : 1,
                    "targets" : [
                        {
                            "POSITION" : 11,
                            "NORMAL" : 12
                        },
                        {
                            "POSITION" : 13,
                            "NORMAL" : 14
                        }
                    ]
                },
                // etc..
            ],

```

Reply:
I get your point. I have updated code. First get the morph target count from the first primitive if the first primitive is exist. Second check other primitives have same number of morph target count.